mlapi/discriminating_algo.py

```diff
@@ -0,0 +1,129 @@
+import statistics
+
+class DiscriminatingAlgo(object):
```

Comment:
Use more precise names for the class and variables:
If it discriminates only for facets, put that in the name

Also those comments next to the variable names indicate that it's critical information, otherwise you wouldn't have written a comment. Delete the comments and put that info in the variable names.

Reply:
I deleted all comments except those related to the 3 parameters because they give more information.

Reply:
also where the 3 filters are located